kornia/geometry/camera/stereo.py

```diff
@@ -0,0 +1,287 @@
+            rectified_right_camera (torch.Tensor): The rectified right camera projection matrix
+                                                    of shape :math:`(B, 3, 4)`
+        """
+        # Ensure correct shapes
+        if rectified_left_camera.ndim != 3:
+            raise StereoException(
+                f"Expected 'rectified_left_camera' to have 3 dimensions. " f"Got {rectified_left_camera.ndim}."
+            )
+
+        if rectified_right_camera.ndim != 3:
+            raise StereoException(
+                f"Expected 'rectified_right_camera' to have 3 dimension. " f"Got {rectified_right_camera.ndim}."
+            )
+
+        if rectified_left_camera.shape[:1] == (3, 4):
+            raise StereoException(
+                f"Expected each 'rectified_left_camera' to be of shape (3, 4)."
+                f"Got {rectified_left_camera.shape[:1]}."
+            )
+
+        if rectified_right_camera.shape[:1] == (3, 4):
+            raise StereoException(
+                f"Expected each 'rectified_right_camera' to be of shape (3, 4)."
+                f"Got {rectified_right_camera.shape[:1]}."
+            )
+
+        # Ensure same devices for cameras.
+        if rectified_left_camera.device != rectified_right_camera.device:
+            raise StereoException(
+                f"Expected 'rectified_left_camera' and 'rectified_right_camera' "
+                f"to be on the same devices."
+                f"Got {rectified_left_camera.device} and {rectified_right_camera.device}."
+            )
+
+        # Ensure same dtypes for cameras.
+        if rectified_left_camera.dtype != rectified_right_camera.dtype:
+            raise StereoException(
+                f"Expected 'rectified_left_camera' and 'rectified_right_camera' to"
+                f"have same dtype."
+                f"Got {rectified_left_camera.dtype} and {rectified_right_camera.dtype}."
+            )
+
+        # Ensure all intrinsics parameters (fx, fy, cx, cy) are the same in both cameras.
+        if not torch.all(torch.eq(rectified_left_camera[..., :, :3], rectified_right_camera[..., :, :3])):
+            raise StereoException(
+                f"Expected 'left_rectified_camera' and 'rectified_right_camera' to have"
+                f"same parameters except for the last column."
+                f"Got {rectified_left_camera[..., :, :3]} and {rectified_right_camera[..., :, :3]}."
+            )
+
+        # Ensure that tx * fx is negative and exists.
+        tx_fx = rectified_right_camera[..., 0, 3]
+        if torch.all(torch.gt(tx_fx, 0)):
+            raise StereoException(f"Expected :math:`T_x * f_x` to be negative." f"Got {tx_fx}.")
+
+    @property
+    def batch_size(self) -> int:
+        r"""Returns the batch size of the storage.
+
+        Returns:
+            int: scalar with the batch size
+        """
+        return self.rectified_left_camera.shape[0]
+
+    @property
+    def fx(self) -> torch.Tensor:
```

Comment:
```suggestion
    def fx_left(self) -> torch.Tensor:
```
to be consistent with the others